spec/spec_helper.rb

```diff
@@ -67,6 +67,22 @@ def expect_shell_command(*command, exitstatus: 0, output: '')
   expect(Open3).to receive(:popen2e).with(*command).and_yield(mock_input, mock_output, mock_thread)
 end
 
+# If the `described_class` of a spec is a `Fastlane::Action` subclass, it runs
+# it with the given parameters.
+#
+def run_described_action(parameters, lane_name = 'test')
+  expect(described_class).to be < Fastlane::Action, "Only call `#{__callee__}` from a spec describing a `Fastlane::Action` subclass."
```

Comment:
I went for a test failure over a `raise` so that we'd get this error as part of the test run result. What do you think?

Reply:
I'm fine either way, though I'd have used `raise` because if called from the wrong place in the test code itself, that's a programmer error from whoever write the spec/test, as opposed to a correct test which goes red and fail an expectation because the SUT / implementation it's testing changed behavior.

To be a failed expectation suggests that the SUT was expected to behave in a certain way, and might have been for a while (making the test green for a while) but suddenly we change something in the implementation code and the test goes red because our expectations of how the SUT is supposed to behave were invalidated. So it's a behavior error and a notification that the behavior changed (after some refactoring of the implementation maybe).

By contrast, calling this method from a test which does not have an `Action` subclass as its SUT is a programmer error in the way you wrote the test, not an expectation of the SUT itself risking changing, so to me a `raise` would make more sense here. But to be honest, that's also not that big a deal 🤷